modules/build.gradle

```diff
@@ -23,6 +23,11 @@ configure(subprojects.findAll { it.parent.path == project.path }) {
     throw new InvalidModelException("Modules cannot contain bin files")
   }
   if (project.file('src/main/config').exists()) {
-    throw new InvalidModelException("Modules cannot contain config files")
+    // The `config` directory is different, because we merge together log4j config files
```

Comment:
@mark-vieira I don't know if this is the right thing to do here. The existing logic in `distribution/build.gradle` will pick up and combine any log4j config files in modules, yet we've banned all config files in modules. I could have just added the relevant logging config to the main config, but I prefer keeping the s3 config close to the relevant source code.

Reply:
I think that is all OBE. You can see we don't enforce the same checks for x-pack/plugin (which are modules) and many of them contain config and bin files as well. I don't think we care about having different rules here for x-pack vs non-xpack. And in any case, if these rules were still applicable they should live in `InternalPluginBuildPlugin` not in the project build script.

In any case I think we can remove this. It makes no sense to forbid having config files, especially when we have [explicit handling for this](https://github.com/elastic/elasticsearch/blob/9a3422e1a6cf519e3fedce396784be2ef48dc7f9/distribution/build.gradle#L172-L177).